packages/core/src/hooks/useLocalStorage.ts

```diff
@@ -1,34 +1,13 @@
 import { useEffect, useState } from 'react'
-
-function getItem(key: string) {
-  if (typeof window === 'undefined') {
-    return null
-  }
-
-  const item = window.localStorage.getItem(key)
-  if (item !== null) {
-    try {
-      return JSON.parse(item)
-    } catch {
-      // ignore error
-    }
-  }
-}
-
-function setItem(key: string, value: any) {
-  if (value === undefined) {
-    window.localStorage.removeItem(key)
-  } else {
-    const toStore = JSON.stringify(value)
-    window.localStorage.setItem(key, toStore)
-    return JSON.parse(toStore)
-  }
-}
+import { useConfig } from './useConfig'
 
 /**
  * @internal Intended for internal use - use it on your own risk
  */
 export function useLocalStorage(key: string) {
+  const {
+    localStorageFunctions: { getItem, setItem },
```

Comment:
There might be a problem with this  destructuring approach, if someone provides a class which relies on `this` for keeping the keys.